### PR TITLE
Ddrayne/edin bus tracker

### PR DESCRIPTION
**Pull request by dan0**
# Description
Added an app for showing upcoming Edinburgh bus times for a given stop

Given a stop ID, it will show up to the next 3 services arriving (in order of closest first), and then show the following arrivals for those services in a scrolling Marquee.

edinburgh_bus

# Copilot

### Summary
🚌🚏📄

This pull request adds a new applet for the Tidbyt device, called Edinburgh Bus, which shows live bus information for a specified stop in Edinburgh. The applet consists of two files: `edinburgh_bus.star`, which contains the main logic and rendering code, and `manifest.yaml`, which provides the applet metadata and configuration options.

> _`Edinburgh Bus` app_
> _Shows live buses on Tidbyt_
> _Autumn leaves fall fast_

### Walkthrough
* Create and implement the Edinburgh Bus applet to show live bus information for a given stop ID in Edinburgh ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139), [link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-6c08700c6542ba96a24a4f3d5b85cfa787f1506587f777a1f4b10c141dce2adeR1-R8))
  * Define the `main` function to fetch and cache the bus data using the `get_bus_data` and `cache_bus_data` functions ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139))
  * Define the `render` function to display the bus data on the Tidbyt device using the `pixlet` library and the `format_bus_data` function ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139))
  * Define the `get_bus_data` function to make an HTTP request to the Transport API using the `requests` library and the `stop_id` configuration option ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139))
  * Define the `cache_bus_data` function to store the bus data in a global variable and update it every 30 seconds using the `time` library ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139))
  * Define the `format_bus_data` function to parse and format the bus data as a list of strings using the `json` library and the `font_size` configuration option ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-0be5810a9b875bc87538c8533a151567b1f26925d1a06f7466d5e20f33da2ef5R1-R139))
  * Add the `manifest.yaml` file to provide metadata and information about the applet, such as its ID, name, summary, description, author, file name, and package name ([link](https://github.com/tidbyt/community/pull/1340/files?diff=unified&w=0#diff-6c08700c6542ba96a24a4f3d5b85cfa787f1506587f777a1f4b10c141dce2adeR1-R8))

### CODE REVIEW

**Comment by dan0** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dan0** [comment]
recheck

**Comment by matslina** [comment]
Hey ddrayne, thanks for the PR. I think the CLA check is failing due to the email address associated to the individual commits not being linked to your github account? Either adding it to the account, or force pushing a commit history with your primary github email address should/could do the trick. Worth a try!

File: apps/edinburghbus/edinburgh_bus.star [hunk 1]
```diff
@@ -0,0 +1,139 @@
+"""
+Applet: Edinburgh Bus
+Summary: Shows the next 3 buses
+Description: Give it an Edinburgh stop ID and it will show the next 3 arrivals for the next 3 services at that stop.
```
Comment (matslina) [inline apps/edinburghbus/edinburgh_bus.star:4]: Based on the description, I'm guessing there should be a `get_schema()` in here somewhere? ([docs](https://tidbyt.dev/docs/reference/schema))

**Review by matslina — CHANGES_REQUESTED** [review]

File: apps/edinburghbus/edinburgh_bus.star (hunk 1 above)
Comment (dan0) [reply apps/edinburghbus/edinburgh_bus.star:4]: Of course, good call!  Will update/

**Comment by dan0** [comment]
Hej Matslina!  I'll take another crack at the email, my tydbit has me using git for the first time in a few years 😃

File: apps/edinburghbus/edinburgh_bus.star (hunk 1 above)
Comment (dan0) [reply apps/edinburghbus/edinburgh_bus.star:4]: Added a config in latest commits - will look at fixing the email issue next

**Comment by dan0** [comment]
Hmm seemed to have messed that up spectacularly - will try again soon 🙃

**Pull request closed without merging**